pkgs/applications/networking/mailreaders/notmuch/default.nix

```diff
@@ -133,6 +134,14 @@ stdenv.mkDerivation rec {
   + lib.optionalString withSfsexp ''
     cp notmuch-git $out/bin/notmuch-git
     wrapProgram $out/bin/notmuch-git --prefix PATH : $out/bin:${lib.getBin git}/bin
+  '' + lib.optionalString withVim ''
+    make -C vim DESTDIR="$out/share/vim-plugins/notmuch" prefix="" install
+    mkdir -p $out/share/nvim
+    ln -s $out/share/vim-plugins/notmuch $out/share/nvim/site
+  '' + lib.optionalString (withVim && withRuby) ''
+    cat >> $out/share/vim-plugins/notmuch/plugin/notmuch.vim << EOF
+      let \$RUBYLIB=\$RUBYLIB . ":$out/${ruby.libPath}/${ruby.system}:${rubyPackages.mail}/lib/ruby/gems/${ruby.version.libDir}/gems/mail-${rubyPackages.mail.version}/lib"
```

Comment:
Hmm I'm surprised there is no shortcut for:

```nix
${rubyPackages.mail}/lib/ruby/gems/${ruby.version.libDir}/gems/mail-${rubyPackages.mail.version}/lib"
```

The python equivalent would have been simply:

```nix
${pythonPackages.mail}/${pythonPackages.python.sitePackages}
```

It would have been nice if `ruby.gems` had provided such an equivalent, but according to my poor `git grep` searches it doesn't seem like a common thing to do - to set `RUBYLIB` to such a search path of both ruby and a few gems.

Either way, it'd be nice to be able to define this variable a bit more abstractly, and generally in the `passthru` like this (untested):

```nix
passthru = {
  # ...
  # Perhaps this should be part of the `ruby.passthru`, or an attribute
  # for all `ruby.gems`.
  getGemRubyLibPath = gem: ${gem.ruby.gemPath}/gems/${gem.gemName}-${gem.version}/lib";
  rubyLibPath = lib.concatStringsSep ":" [
    "$out/${ruby.libPath}/${ruby.system}"
  ] ++ lib.map finalAttrs.passthru.getGemRubyLibPath [
    ruby.gems.mail
  ];
};
```

Note how `getGemRubyLibPath` does what you wanted to do much more elegantly. With the above, you can access the `RUBYLIB` path you want with:

```suggestion
      let \$RUBYLIB=\$RUBYLIB . ":${finalAttrs.passthru.rubyLibPath}"
```

You'd also have to transition to `finalAttrs` (like [this commit for instance](https://github.com/NixOS/nixpkgs/commit/c1f5835e6ba8b2df709d51fa0e81b1f51fd7105c)) in a separate commit, before this commit.

Reply:
Thanks for the suggestions!

> You'd also have to transition to finalAttrs (like [this commit for instance](https://github.com/NixOS/nixpkgs/commit/c1f5835e6ba8b2df709d51fa0e81b1f51fd7105c)) in a separate commit, before this commit.

Can I ask why does this need to be in a separate commit?

Reply:
Okay, I've changed the approach a bit to use `buildEnv`.